jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ProjectProperties.java

```diff
@@ -46,21 +48,38 @@ String getMainClass(@Nullable String mainClass) {
     if (mainClass == null) {
       mainClass = getMainClassFromJarTask();
       if (mainClass == null) {
-        throw new GradleException(
-            HelpfulSuggestionsProvider.get("Could not find main class specified in a 'jar' task")
-                .suggest("add a `mainClass` configuration to jib"));
+        gradleBuildLogger.info(
+            "Could not find main class specified in a 'jar' task; attempting to "
+                + "infer main class.");
+        try {
+          mainClass =
+              MainClassFinder.findMainClass(
+                  project
```

Comment:
This should use `GradleSourceConfiguration#getClassesFiles`, which will be the files that go into the container.

Reply:
This returns a list of Paths, how would I use it to get the root directory?